oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+files as they are copied to their locations amongst the code. The atlas.yml file
+will also support selecting which languages to be included in an Open edX deployment. The
+tool will have to be used/ran as part of the setup of a repository, whether for
+development or deployment.
+
+Impacts
+*******
+
+Impact on Translators
+=====================
+
+As we approach the end of the translation upgrade process, we will need to tactically
+move from multiple Transifex projects to a single project. This will require coordination
+with our translators to ensure that moving forward they are providing translations in the
+right place.
+
+Impact on Site Operators
+========================
+
+Currently the translation files for any given service or library is stored at the same
+place as the code, which has generally simplified the deployment story in the past. With
+this change, the translations files will move to their own repository. As we deprecate
+the old translations files, the relevant deployment tooling will need to be updated to
+pull down the translations from the new repository as a part of the deployment process.
+This will impact both the old Ansible based tooling as well as any new Docker based
+tooling.
+
+Impact on Developers
+====================
+
+While it won’t directly impact the day-to-day workflow of developers (unless you are
+developing or testing with translation files), due to the same reasons that we impact
+site operators (new translations location), we will have to update development tools as
+well. In addition, we will create new instructions for developers on how to enable
+translations for a new service/repo when it comes online.
+
+Locations
+*********
+
+Dumps of the translation/localization files from Transifex for the Open edX Release
+project already exist in a repository with the name of openedx/openedx-i18n. A new
+repository named openedx/openedx-translations will be similarly structured, but it will
+contain the translation files for all repositories within openedx. The
+`Transifex GitHub App`_ will be installed in the openedx organization. Similar to how the
+Build-Test-Release Working Group creates a new branch for each new named release of
+edx-platform, translation releases will also be kept in branches corresponding to
+edx-platform releases.
+
+Rejected Alternatives
+*********************
+
+Rewriting the Current Tooling for the New API
+=============================================
+
+The source code for the edx-transifex-bot can be found in `ecommerce-scripts`_. We could
+rewrite the current tooling to try to solve the problems encountered in the last two Open
+edX releases and upgrade to the new API, but this approach is a patch-up job that will
+not address several other issues mentioned and would have to be undertaken by the
+community member with exclusive access to the legacy infrastructure currently running the
+edx-transifex-bot.
+
+Making a Transifex Project for Each Repository
```

Comment:
Possibly, but then we have to consider that a mostly non-technical Translation WG has to learn how to use the Transifex API. And then Translators may have to be part of 100+ teams to translate files for each repository, and learn to navigate the hundreds of projects to see their progress.

Reply:
@pshiu for this and some other comments, it would be useful to spell out your use cases (both ones you currently follow and new ones you're proposing - and for the latter, justification to the need)

Reply:
@sarina Sorry for the confusion! My comment was meant to address the rejected alternative of making a Transifex project for each repository. The suggestion was that perhaps tooling could be created to remedy the concerns that caused this alternative be rejected. (For example, perhaps a bot that auto-adds and removes translators from all relevant projects.)

@Carlos-Muniz I thought that even in a consolidated repository scenario, translators still have to navigate one [Transifex Resource](https://help.transifex.com/en/articles/6656105-introduction-to-projects-and-resources) per original source repository (resulting in the same problem)?

Reply:
@pshiu Translators are capable of navigating between Transifex Resources (Translation files) within one Transifex Project, as they do that already with the current Translation Project "edx-platform". File structures are flattened within a transifex project, so they will not have to search through directories to find the resources they are to translate. But having to look between 100+ Transifex projects for 1-2 resources each project, and then expecting Transifex Admin to be able to track the progress of each of the languages for each of the projects becomes a far more labor intensive alternative. 

Tooling + a dashboard to track progress would have to be made at the very least. And with all those extra moving parts, we may find ourselves with a pile of spaghetti that would be very difficult to untangle.

Reply:
@Carlos-Muniz After diving into Transifex's UI, here's what I now understand:
* Transifex's string search [[example search](https://www.transifex.com/open-edx/search/?q=translation_text%3Abuscar+target_language%3Aes_419), [docs](https://help.transifex.com/en/articles/6240799-search-strings)] allows for cross-project string searches.
* Dashboards that document language progress across projects [[all projects dashboard](https://www.transifex.com/open-edx/), [all projects language dashboard](https://www.transifex.com/open-edx/dashboard/all_projects/es_419/)] seem to be available.
* The Transifex editor supports an "All Resources" option, but not an "All Projects" option.

That last bullet means (like you said) that to translate or review a string across projects, a translator will either have to:

1. Click on the editor links in the string search, which loads an editor in a new tab, or
2. Know which project a string belongs to and switch to that project in the editor's project dropdown menu.

Both are tedious for the translator and a bad user experience. The fact that the Transifex editor does not support editing strings across multiple projects is (at least for me) the winning argument justifying the consolidation of all translation strings into a single Transifex project.

I didn't understand this when I first read the OEP, so I really appreciate your time in enlightening me. Thanks for your help.